api_tests/nodes/views/test_node_addons.py

```diff
@@ -0,0 +1,747 @@
+# -*- coding: utf-8 -*-
+import abc
+import mock
+from nose.tools import *  # flake8: noqa
+
+from api.base.settings.defaults import API_BASE
+from framework.auth.core import Auth
+from website.util.permissions import READ, WRITE, ADMIN
+
+from tests.base import ApiAddonTestCase
+from tests.factories import AuthUserFactory
+from website.addons.box.tests.factories import BoxAccountFactory, BoxNodeSettingsFactory
+from website.addons.dataverse.tests.factories import DataverseAccountFactory, DataverseNodeSettingsFactory
+from website.addons.dropbox.tests.factories import DropboxAccountFactory, DropboxNodeSettingsFactory
+from website.addons.github.tests.factories import GitHubAccountFactory, GitHubNodeSettingsFactory
+from website.addons.googledrive.tests.factories import GoogleDriveAccountFactory, GoogleDriveNodeSettingsFactory
+from website.addons.mendeley.tests.factories import MendeleyAccountFactory, MendeleyNodeSettingsFactory
+from website.addons.s3.tests.factories import S3AccountFactory, S3NodeSettingsFactory
+from website.addons.zotero.tests.factories import ZoteroAccountFactory, ZoteroNodeSettingsFactory
+
+
+class NodeAddonListMixin(object):
+    def set_setting_list_url(self):
+        self.setting_list_url = '/{}nodes/{}/addons/?page[size]=100'.format(
+            API_BASE, self.node._id
+        )
+
+    def get_response_for_addon(self, response):
+        for datum in response.json['data']:
+            if datum['id'] == self.short_name:
+                return datum['attributes']
+        return None
+
+    def test_settings_list_GET_enabled(self):
+        wrong_type = self.should_expect_errors()
+        res = self.app.get(
+            self.setting_list_url,
+            auth=self.user.auth)
+
+        addon_data = self.get_response_for_addon(res)
+        if not wrong_type:
+            assert_equal(self.account_id, addon_data['external_account_id'])
+            assert_equal(self.node_settings.has_auth, addon_data['node_has_auth'])
+            assert_equal(self.node_settings.folder_id, addon_data['folder_id'])
+        if wrong_type:
+            assert_equal(addon_data, None)
+
+    def test_settings_list_GET_disabled(self):
+        wrong_type = self.should_expect_errors()
+        try:
+            self.node.delete_addon(self.short_name, auth=self.auth)
+        except ValueError:
+            # If addon was mandatory -- OSFStorage
+            pass
+        res = self.app.get(
+            self.setting_list_url,
+            auth=self.user.auth,
+            expect_errors=wrong_type)
+        addon_data = self.get_response_for_addon(res)
+        if not wrong_type:
+            assert_equal(addon_data['external_account_id'], None)
+            assert_false(addon_data['node_has_auth'])
+        if wrong_type:
+            assert_equal(addon_data, None)        
+
+    def test_settings_list_raises_error_if_not_GET(self):
```

Comment:
I'll let you make the call, but IMO it's bad testing semantics to combine these into a single case.